non-config/trader/IEL_b1.py

```diff
@@ -74,7 +75,10 @@ def __init__(self, account, email, password, marketplace_id, valuations):
         # Updated after bot has been initialised and can retrieve these
         # values from the market
         self.valuations = valuations
-        self.max_units = 0
+        # additional 0 is added to the valuation array fix the index error
+        self.valuations.append(0)
```

Comment:
I'm not entirely sure how the index error occurs -- I assume it's when the bot has the maximum number of units. But the bot seems to work for me without having to append an extra 0.

Reply:
That's correct: when a robot has the max number of units and still submits a buy order, the valuation array size is exceeded. The error is fixed when an extra 0 is appended. This is something that both Prof. Ledyard and I have observed when we run simulations.